lib/hacluster.pm

```diff
@@ -1137,4 +1138,47 @@ sub check_iscsi_failure {
     }
 }
 
+=head3 cluster_check_status_by_regex
+
+Remotely check 'crm status' output against regular expression
+=over 3
+
+=item B<SHOW_CLUSTER_STATUS> - Output from 'crm status' command
+
+=item B<REGULAR_EXPRESSION> - Regular expression to match the text to find
+
+=back
+=cut
+
+sub cluster_check_status_by_regex {
+    my ($show_cluster_status, $regular_expression) = @_;     
+    croak 'No crm status output' if (!$show_cluster_status);
+    croak 'No regular expression provided' unless (ref $show_cluster_status eq 'Regexp');
+  
+    my $done = 0;   
+    
+    for my $line (split("\n", $show_cluster_status)) {
+        if ($line =~ $regular_expression) {
```

Comment:
Thinking of ways to avoid nested loops, one possible solution could be to do:
```
if (($line =~ $regular_expression) && ($line =~ /(Stopped|Failed|Pending|Blocked)/))
```
That way you can check both conditions in one go.

Reply:
good idea